src/app/tech-carbon-estimator/tech-carbon-estimator.component.ts

```diff
@@ -35,9 +39,40 @@ export class TechCarbonEstimatorComponent {
 
   constructor(
     private estimationService: CarbonEstimationService,
-    private changeDetector: ChangeDetectorRef
+    private changeDetector: ChangeDetectorRef,
+    private ref: ElementRef
   ) {}
 
+  ngOnInit() {
+    this.insertShadowStylesLink();
+  }
+
+  private insertShadowStylesLink() {
+    // Reasons for this approach:
+    // 1. Angular global injection would insert the tag in the page root, so we disabled it.
+    // 2. Component `styleUrl` wouldn't allow us to vary stylesheets based on build configurations.
+
+    let stylesLink = this.createShadowStylesLink('styles.css');
+    let googleFontsLink = this.createShadowStylesLink('https://fonts.googleapis.com/icon?family=Material+Icons+Outlined');
+
+    this.ref.nativeElement.shadowRoot.appendChild(googleFontsLink);
+    this.ref.nativeElement.shadowRoot.appendChild(stylesLink);
+  }
+
+  private createShadowStylesLink(styleHref: string) {
+
+    const isAbsoluteUrl = styleHref.startsWith('http://') || styleHref.startsWith('https://');
```

Comment:
It may well be simpler to deal with this path wrangling in the calling method - it's only our own styles that needs anything